src/Microsoft.AspNetCore.Authentication/Events/FailureContext.cs

```diff
@@ -9,10 +9,14 @@ namespace Microsoft.AspNetCore.Authentication
     /// <summary>
     /// Provides failure context information to handler providers.
     /// </summary>
-    public class FailureContext : BaseControlContext
+    public class FailureContext : HandleRequestContext<RemoteAuthenticationOptions>
```

Comment:
Yes. The associated event is named `RemoteFailure`. Do you want me to change the context name?

On a related note, I've never been a huge fan of the RAT concept because it hides the protocol details, which prevents us from exposing things like `error`/`error_description`/`error_uri` `OpenIdConnectMessage` in this kind of events (and that's why we have a separate event for OIDC, `AuthenticationFailed`).

I suppose removing RAT would be unacceptable for you? :sweat_smile: